### ISSUE TITLE
Code as provided does not compile

### ISSUE THREAD

**Issue by RobertBrown3** [Avansai/properties-file#21]
I have written a simple Typescript function intended to read the contents of a properties file. The code i question is test code and is provided below:

```ts
import { readFileSync } from 'node:fs'
import { Properties } from 'properties-file'

export const readProperties = () => {
    const props = new Properties(readFileSync('client.properties'));

    props.collection.forEach(property =>{
        console.log('Property: '+property.key+' '+property.value);
    });
}
```

When I attempt to compile this code, I get the error messages included below. There is clearly something wrong (missing?) with the properties-file library. Until the problem is fixed, this library is unusable.

Can this problem be fixed soon? Or perhaps there is a workaround that can make this code compile?

```sh
------------------------------------------Error Messages (3 errors)----------------------------------------------------------------

Compiled with problems:
×
ERROR in ./src/Network/PropManager.ts 6:20-30
export 'Properties' (imported as 'Properties') was not found in 'properties-file' (possible exports: getProperties)
ERROR in ./node_modules/properties-file/lib/esm/index.js 1:0-42
Module not found: Error: Can't resolve './properties' in 'C:\Research\Project 30\demoui\node_modules\properties-file\lib\esm'
Did you mean 'properties.js'?
BREAKING CHANGE: The request './properties' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
ERROR in node:fs
Module build failed: UnhandledSchemeError: Reading from "node:fs" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
    at C:\Research\Project 30\demoui\node_modules\webpack\lib\NormalModule.js:838:25
    at Hook.eval [as callAsync] (eval at create (C:\Research\Project 30\demoui\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (C:\Research\Project 30\demoui\node_modules\tapable\lib\Hook.js:18:14)
    at Object.processResource (C:\Research\Project 30\demoui\node_modules\webpack\lib\NormalModule.js:835:8)
    at processResource (C:\Research\Project 30\demoui\node_modules\loader-runner\lib\LoaderRunner.js:220:11)
    at iteratePitchingLoaders (C:\Research\Project 30\demoui\node_modules\loader-runner\lib\LoaderRunner.js:171:10)
    at runLoaders (C:\Research\Project 30\demoui\node_modules\loader-runner\lib\LoaderRunner.js:398:2)
    at NormalModule._doBuild (C:\Research\Project 30\demoui\node_modules\webpack\lib\NormalModule.js:825:3)
    at NormalModule.build (C:\Research\Project 30\demoui\node_modules\webpack\lib\NormalModule.js:969:15)
    at C:\Research\Project 30\demoui\node_modules\webpack\lib\Compilation.js:1374:12
    at NormalModule.needBuild (C:\Research\Project 30\demoui\node_modules\webpack\lib\NormalModule.js:1262:26)
    at Compilation._buildModule (C:\Research\Project 30\demoui\node_modules\webpack\lib\Compilation.js:1355:10)
    at C:\Research\Project 30\demoui\node_modules\webpack\lib\util\AsyncQueue.js:305:10
    at Hook.eval [as callAsync] (eval at create (C:\Research\Project 30\demoui\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncQueue._startProcessing (C:\Research\Project 30\demoui\node_modules\webpack\lib\util\AsyncQueue.js:295:26)
    at AsyncQueue._ensureProcessing (C:\Research\Project 30\demoui\node_modules\webpack\lib\util\AsyncQueue.js:282:12)
    at process.processImmediate (node:internal/timers:476:21)
```

**Comment by nbouvrette** [comment]
hey @RobertBrown3, could you provide an example repository to reproduce your issue?

I added dual support for CommonJS and ESM module earlier this week and your issue might be related to the way the ESM modules are built using `tsc` as documented here: https://github.com/microsoft/TypeScript/issues/16577

If you are able to provide me an example it would help me pinpoint the exact problem.

Also you might have noticed but you seem to have another issue here::

`Module build failed: UnhandledSchemeError: Reading from "node:fs" is not handled by plugins (Unhandled scheme).`

**Comment by nbouvrette** [comment]
Hi @RobertBrown3 - I was able to reproduce the issue and fixed it in version [3.4.1](https://github.com/Avansai/properties-file/releases/tag/3.4.1) - details on this commit: https://github.com/Avansai/properties-file/commit/98f53c8d5f37f2c6a16a3f8e814a993d387c75d6

I'm going to close this issue for now - let me know if it didn't resolve your original issue.
